libs/api_common/device_helper.cpp

```diff
@@ -27,21 +27,35 @@ typedef struct _async_ioctl_completion_context
 } async_ioctl_completion_context_t;
 
 // Handle used for synchronous calls to the driver.
```

Comment:
Would it make sense to implement `_sync_device_handle` and `_async_device_handle` as instances of a new "device_handle" class (w/constr(), destr() and get()), eventually with an extra-safety on the thread-detach system call (even though the compiler should guarantee the destructor to be called)?